sheerwater_benchmarking/forecasts/fuxi.py

```diff
@@ -0,0 +1,230 @@
+"""Interface for FuXi forecasts."""
+
+import os
+import glob
+import shutil
+import dask
+import xarray as xr
+import numpy as np
+import pandas as pd
+import py7zr
+
+from huggingface_hub import hf_hub_download
+from huggingface_hub.utils import EntryNotFoundError
+
+from sheerwater_benchmarking.utils.secrets import huggingface_read_token
+from sheerwater_benchmarking.utils import (dask_remote, cacheable,
+                                           apply_mask, clip_region,
+                                           lon_base_change,
+                                           target_date_to_forecast_date,
+                                           shift_forecast_date_to_target_date, lead_to_agg_days, roll_and_agg)
+
+
+@dask_remote
+@cacheable(data_type='array', cache_args=['date'])
+def fuxi_single_forecast(date):
+    """Download a single forecast from the FuXi dataset."""
+    token = huggingface_read_token()
+
+    date = str(date)
+    date = date.replace('-', '')
+    fname = date + '.7z'
+    target_path = f"./{fname.replace('.7z', '')}"
+
+    # Download the file
+    try:
+        path = hf_hub_download(repo_id="FudanFuXi/FuXi-S2S", filename=fname, repo_type="dataset", token=token)
+    except EntryNotFoundError:
+        print("Skipping invalid forecast date")
+        return None
+
+    # unzip the file
+    with py7zr.SevenZipFile(path, mode='r') as zip:
+        print(f"Extracting {path} to {target_path}")
+        zip.extractall(path=target_path)
+
+    files = glob.glob(f"{target_path}/**/*.nc", recursive=True)
+
+    def pre(ds):
+        ff = ds.encoding["source"]
+        member = ff.split('/')[-2]
+        ds = ds.assign_coords(member=member)
+        ds = ds.expand_dims(dim='member')
+        return ds
+
+    # Transform and drop dataa
+    print("Opening dataset")
+    ds = xr.open_mfdataset(files, engine='netcdf4', preprocess=pre)
+    return ds
+
+    ds = ds['__xarray_dataarray_variable__'].to_dataset(dim='channel')
+
+    variables = [
+        'tp',
+        't2m',
+        'd2m',
+        'sst',
+        'ttr',
+        '10u',
+        '10v',
+        'msl',
+        'tcwv'
+    ]
+
+    print(ds)
+
+    ds = ds[variables]
+    ds = ds.compute()
+
+    # Delete the files
+    link = os.path.realpath(path)
+    os.remove(path)
+    os.remove(link)
+    shutil.rmtree(target_path)
+
+    return ds
+
+@dask_remote
+@cacheable(data_type='array', cache_args=[], timeseries='time',
+           chunking={'lat': 121, 'lon': 240, 'lead_time': 14, 'time': 2, 'member': 51})
+def fuxi_raw(start_time, end_time, delayed=False):
+    """Combine a range of forecasts with or without dask delayed."""
+    dates = pd.date_range(start_time, end_time)
```

Comment:
Is this a daily timeseries? can you add a comment that says that's the deafult or specify?